keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,504 @@
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ProcMountType
+  - Components depending on the feature gate: kubelet, kube-apiserver (whole cluster must have `ProcMountType` feature gate enabled)
+
+###### Does enabling the feature change any default behavior?
+
+No, only allows a user to access the Unmasked ProcMountType
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. This can be done by removing the feature gate from all kube-apiservers, and then restarting the nodes (to restart the pods that originally got
+access to Unmasked ProcMountType).
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing. All pods that requested Unmasked ProcMountType had it stripped from the request because the feature was off. All other workloads remain unchanged.
+Recreating pods that requested Unmasked will result in it being applied as expected, once the feature is on again.
+
+###### Are there any tests for feature enablement/disablement?
+
+No
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Inconsistent feature gate enablement is the only way. All supported versions of CRIs (CRI-O and containerd) have support for this feature.
+
+###### What specific metrics should inform a rollback?
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+They will be, specifically cases where the Kubelet is restarted.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+`kubectl get pods --all-namespaces -o jsonpath="{range .items[*]}{.metadata.name}{' '}{.spec.containers[*].securityContext.procMount}{'\n'}{end}"  | grep -i unmasked`
+Will print all pods that has an Unmasked ProcMountType, along with the pod name.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [x] Other (treat as last resort)
+  - Details: Container created with the Unmasked ProcMountType have paths [here](https://github.com/kubernetes/kubernetes/blob/964529b/pkg/securitycontext/util.go#L193) as writable, not read only.
+    - "/proc/asound",
+	- "/proc/acpi",
+	- "/proc/kcore",
+	- "/proc/keys",
+	- "/proc/latency_stats",
+	- "/proc/timer_list",
+	- "/proc/timer_stats",
+	- "/proc/sched_debug",
+	- "/proc/scsi",
+	- "/sys/firmware",
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
```

Comment:
Don't we have a reasonable SLO for that kubelet_pod_start_sli_duration_seconds metric? Something like a few seconds? At least you can describe that the metric should maintain values from before the upgrade, for example.

Reply:
see https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md#kubernetes-slisslos

Reply:
updated!